### PR TITLE
task#3_2_fetch_data_home_screen

### PR DESCRIPTION
**Pull request by maigiang08**

### CODE REVIEW

File: IOSMarvel/IOSMarvel/Base Classes/BaseViewController.swift [hunk 1]
```diff
@@ -24,4 +25,15 @@ class BaseViewController: UIViewController {
...
     }
+
+    func showLoadingOnParent() {
+        let size = CGSize(width: 30, height: 30)
+        startAnimating(size, message: "", messageFont: nil, type: NVActivityIndicatorType(rawValue: 14)!,
+                       color: .black, padding: 0, displayTimeThreshold: 0,
+                       minimumDisplayTime: 0, backgroundColor: .clear, textColor: .black)
+    }
+
+    func hinLoading() {
```
Comment (daolq3012) [inline IOSMarvel/IOSMarvel/Base Classes/BaseViewController.swift]: hinLoading -> hideLoading

File: IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift [hunk 2]
```diff
@@ -49,6 +57,57 @@ class HomeViewController: BaseViewController {
...
+            }
+        }
+    }
+
+    func loadMoreData() {
+        if isLoading {
+            return
+        }
+        self.showLoadingOnParent()
+        characterRepository.getListCharacter(limit: self.page * 10) { (result) in
```
Comment (daolq3012) [inline IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift]: Pls define ``10`` as a constant variable

File: IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift [hunk 3]
```diff
@@ -49,6 +57,57 @@ class HomeViewController: BaseViewController {
...
+                if self.characterList.count < self.limit {
+                    self.isLoadMore = false
+                } else {
+                    self.isLoadMore = true
+                }
+                self.characterCollectionView.reloadData()
+                self.page += 1
+                self.hinLoading()
+            case .failure(let error):
+                print("error: ", error!)
```
Comment (daolq3012) [inline IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift]: show Alert when request API fail

File: IOSMarvel/IOSMarvel/Base Classes/BaseViewController.swift (hunk 1 above)
Comment (maigiang08) [reply IOSMarvel/IOSMarvel/Base Classes/BaseViewController.swift]: Vâng ạ

File: IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift (hunk 2 above)
Comment (maigiang08) [reply IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift]: Vâng ạ

File: IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift (hunk 3 above)
Comment (maigiang08) [reply IOSMarvel/IOSMarvel/Modules/Home/HomeViewController.swift]: Vâng ạ

**Pull request merged**